man/fi_endpoint.3.md

```diff
@@ -520,6 +520,19 @@ The following option levels and option names and parameters are defined.
   that applications that want to override the default MIN_MULTI_RECV
   value set this option before enabling the corresponding endpoint.
 
+- *FI_OPT_FI_HMEM_P2P - int*
+: Defines how the provider should handle peer to peer FI_HMEM transfers for
+  this endpoint. The default option is to require peer to peer support. Valid
+  values defined in fi_endpoint.h are:
+	* FI_HMEM_P2P_ENABLED: Peer to peer support is required and must be used for
+	  FI_HMEM transfers.
+	* FI_HMEM_P2P_PREFERRED: Peer to peer support should be used if
+	  available, but the provider may choose to copy the data to initiate
+	  the transfer if peer to peer support is unavailable.
+	* FI_HMEM_P2P_DISABLED: Peer to peer support should not be used.
+: fi_setopt() will return -FI_EINVAL if the mode requested cannot be supported
+  by the provider.
```

Comment:
EINVAL indicates a invalid value was specified.  Use OPUNOTSUPP or ENOSYS to indicate the operation isn't supported